src/string.rs

```diff
@@ -140,6 +190,43 @@ impl<const N: usize> String<N> {
         self.vec.extend_from_slice(string.as_bytes())
     }
 
+    /// Append a given UTF-16-encoded slice onto the end of this `String`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use heapless::String;
+    ///
+    /// // 𝄞music
+    /// let mut s: String<10> = String::new();
+    /// let v = &[0xD834, 0xDD1E, 0x006d, 0x0075,
+    ///           0x0073, 0x0069, 0x0063];
+    /// s.push_utf16(v).unwrap();
+    /// assert_eq!(s, "𝄞music");
+    ///
+    /// // 𝄞mu<invalid>ic
+    /// let mut s: String<10> = String::new();
+    /// let v = &[0xD834, 0xDD1E, 0x006d, 0x0075,
+    ///           0xD800, 0x0069, 0x0063];
+    /// assert!(s.push_utf16(v).is_err());
+    /// ```
+    pub fn push_utf16(&mut self, v: &[u16]) -> Result<(), Error> {
+        for c in decode_utf16(v.iter().cloned()) {
```

Comment:
nit: could you please use a qualified function call here? i.e. `char::decode_utf16`